utils/firebase/userData.ts

```diff
@@ -0,0 +1,27 @@
+import { doc, setDoc } from "firebase/firestore";
+import { UserDoc } from "models/users";
+import { userRef } from "./init";
+
+export const addToUserOwnedLists = async (listID: string, userData: UserDoc) => {
+
+    const newUserData: UserDoc = {
+        ...userData,
+        lists: [...userData.lists, listID]
+    }
```

Comment:
I actually just figured out how to do this at work - [Firestore has functions we can use to add / remove to array fields without having to copy then modify the OG data](https://cloud.google.com/firestore/docs/manage-data/add-data#update_elements_in_an_array).